cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -60,9 +60,18 @@ std::vector<mapping::SubmapId> SparsePoseGraph::GrowSubmapTransformsAsNeeded(
   if (insertion_submaps.size() == 1) {
     // If we don't already have an entry for the first submap, add one.
     if (submap_data.SizeOfTrajectoryOrZero(trajectory_id) == 0) {
-      optimization_problem_.AddSubmap(
-          trajectory_id,
-          sparse_pose_graph::ComputeSubmapPose(*insertion_submaps[0]));
+      transform::Rigid2d submap_pose =
+          sparse_pose_graph::ComputeSubmapPose(*insertion_submaps[0]);
+      if (initial_trajectory_poses_.find(trajectory_id) !=
+          initial_trajectory_poses_.end()) {
+        const InitialTrajectoryPose& initial_pose =
+            initial_trajectory_poses_.at(trajectory_id);
+        submap_pose = transform::Project2D(
```

Comment:
Isn't this too late to take `initial_pose` into account? The node was already added at the wrong place in `AddScan()` before this submap was added? Should this be handled in `ComputeLocalToGlobalTransform()`?